wallets/metamask/src/utils/waitFor.ts

```diff
@@ -21,6 +22,60 @@ export const waitForSelector = async (selector: string, page: Page, timeout: num
   }
 }
 
+export const waitForMetaMaskLoad = async (page: Page) => {
+  await Promise.all(
+    LoadingSelectors.loadingIndicators.map(async (selector) => {
+      await waitForSelector(selector, page, 5000)
+    })
+  )
+    .then(() => {
+      console.log('All loading indicators are hidden')
+    })
+    .catch((error) => {
+      console.error('Error: ', error)
+    })
+
+  return page
+}
+
+export const waitForMetaMaskWindowToBeStable = async (page: Page) => {
+  await waitForMetaMaskLoad(page)
+  if ((await page.locator(ErrorSelectors.loadingOverlayErrorButtons).count()) > 0) {
+    const retryButton = await page.locator(ErrorSelectors.loadingOverlayErrorButtonsRetryButton)
+    await retryButton.click()
+    await waitForSelector(LoadingSelectors.loadingOverlay, page, 300)
```

Comment:
why sometimes it's 5000 and sometimes only 300ms?

Reply:
Resolved by DEFAULT_TIMEOUT